internal/cephfs/core/clone.go

```diff
@@ -19,43 +19,42 @@ package core
 import (
 	"context"
 	"errors"
+	"fmt"
 
 	cerrors "github.com/ceph/ceph-csi/internal/cephfs/errors"
 	"github.com/ceph/ceph-csi/internal/util/log"
+
+	"github.com/ceph/go-ceph/cephfs/admin"
 )
 
 // cephFSCloneState describes the status of the clone.
-type cephFSCloneState string
+type cephFSCloneState struct {
+	state    admin.CloneState
+	errno    string
+	errorMsg string
+}
 
 const (
-	// CephFSCloneError indicates that fetching the clone state returned an error.
-	CephFSCloneError = cephFSCloneState("")
-	// CephFSCloneFailed indicates that clone is in failed state.
-	CephFSCloneFailed = cephFSCloneState("failed")
-	// CephFSClonePending indicates that clone is in pending state.
-	CephFSClonePending = cephFSCloneState("pending")
-	// CephFSCloneInprogress indicates that clone is in in-progress state.
-	CephFSCloneInprogress = cephFSCloneState("in-progress")
-	// CephFSCloneComplete indicates that clone is in complete state.
-	CephFSCloneComplete = cephFSCloneState("complete")
-
 	// SnapshotIsProtected string indicates that the snapshot is currently protected.
 	SnapshotIsProtected = "yes"
 )
 
-// toError checks the state of the clone if it's not cephFSCloneComplete.
-func (cs cephFSCloneState) toError() error {
-	switch cs {
-	case CephFSCloneComplete:
+// CephFSCloneError indicates that fetching the clone state returned an error.
+var CephFSCloneError = cephFSCloneState{}
+
+// ToError checks the state of the clone if it's not cephFSCloneComplete.
+func (cs cephFSCloneState) ToError() error {
+	switch cs.state {
+	case admin.CloneComplete:
 		return nil
-	case CephFSCloneError:
-		return cerrors.ErrInvalidClone
-	case CephFSCloneInprogress:
+	case CephFSCloneError.state:
+		return fmt.Errorf("%w: %s (%s)", cerrors.ErrInvalidClone, cs.errorMsg, cs.errno)
```

Comment:
is it for an empty state? this looks confusing can you change this to `case ""` or define a const of empty state?

Reply:
`CephFSCloneError` is the const, and it is documented? I think this is clearer than comparing to `""`.